core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -290,4 +301,22 @@ public void process(RestRequest request, RestChannel channel, RestFilterChain fi
             executeHandler(request, channel);
         }
     }
+
+    private static final class RestHandlerHolder {
```

Comment:
would it make more sense to add a method to the `RestHandler` interface `default boolean canTripCircuitBreaker() { return true; }` that would safe us all the extra indirection and is simple to override?